src/render/tutorials/chrome-plugin/2_2.html.md

```diff
@@ -0,0 +1,495 @@
+
+```hocon
+payments-task {
+  action = fetch-user-info
+
+  on._success {
+    actions = [
+      {
+        action = fetch-payment-providers
+      },
+      {
+        action = fetch-offers
+      }
+    ]
+
+    on._success {
+      action = te-pebble
+    }
+  }
+}
+```
+
+After `fetch-user-info` succeeds the composite is started to be processed. It calls `fetch-payment-providers` and `fetch-offers` in parallel. When both of them succeeds the pebble node starts to be processed. We don't need to change any actions, because we've only modified the structure of the graph and not what we do with data.
+
+When we load the page now nothing seems to have changed. However, when we open the chrome extension we can see that the second fragment looks like this:
+
+![Composite](/img/blog/chrome-extension/composite.png)
+
+The white circle represents the beginning of a composite. The second circle is where the composite ended. All subtasks (nodes in the composite) ended with success thus the composite did too. That's why the second circle is green.
+
+This is a good moment to introduce the perfomance tab. When you open the tab called "Perfomance View" you'll see a timeline of node execution:
+
+![Timeline](/img/blog/chrome-extension/timeline.png)
+
+By looking at this it's clear that `fetch-payment-providers` and `fetch-offers` executed simultanously and that the next node (`te-pebble`) executed only after both of them finished.
+
+This is far better from what we had before. There is still room for improvement though. Right now we don't handle any errors. In a real application errors are unavoidable. Let's say that the service that we call to retrieve offers has a long response time. We can simulate it by using a mock `/timeout`. It responds with the same data as `/offers` but it takes three seconds to do so. Swap the endpoint in the `fetch-offers` action:
+
+```hocon
+fetch-offers {
+  factory = http
+  config.endpointOptions {
+    path = /timeout
+    domain = webapi
+    port = 8080
+  }
+}
+```
+
+When we load the page now it should take about three seconds. If we open the Knot.x extension we can immediatly spot on the fragment list that the second fragment executed for a long time. When we open the fragments performance view we can clearly see that the problem lays in the composite and specificaly in the `fetch-offers` node.
+
+What if such a long time is unacceptable? One approach would be to define timeout in the `fetch-offers` action. However, we can do better. Knot.x provides us with a built-in implementation of a [Circuit Breaker Pattern](https://en.wikipedia.org/wiki/Circuit_breaker_design_pattern). It does so by a [Circuit Breaker Factory](TODO-find-a-link-to-documentation) that acts as a wrapper. Let's see how it can be configured:

```

Comment:
```suggestion
What if such a long time is unacceptable? One approach would be to define a timeout in the `fetch-offers` action. However, we can do better. Knot.x provides us with a built-in implementation of a [Circuit Breaker Pattern](https://en.wikipedia.org/wiki/Circuit_breaker_design_pattern). It does so by a [Circuit Breaker Factory](TODO-find-a-link-to-documentation) that acts as a wrapper. Let's see how it can be configured:
```